### ISSUE TITLE
环境问题

### ISSUE THREAD

**Issue by Gavingx** [IDEA-CCNL/Fengshenbang-LM#133]
你好，照着官方给的fengshen/requirement.txt环境装好包后，跑不通。我的环境是：
python==3.8.10
pytorch-lightning==1.6.3
torch==1.9.1+cu111
transformers==4.22.1
datasets==2.4.0
deepspeed==0.5.10
jieba-fast==0.53
jieba==0.42.1
protobuf==3.20.1
尝试换了transformers、pytorch lightning版本都不行。请问能提供个可以跑通的环境吗?或者提供一个docker镜像也是极好的。

**Comment by GGGGGGXY** [comment]
docker镜像下周会给到，可以先看一下你的报错吗，我看看是啥环境问题

**Comment by Gavingx** [comment]
> docker镜像下周会给到，可以先看一下你的报错吗，我看看是啥环境问题

transformers==4.19.0可以跑通。不过出现了新的问题，按照https://zhuanlan.zhihu.com/p/539870077中所述，40G的显卡就可以完成二郎神的微调。但是我微调的时候4张卡都占满了，然后还报错：CUDA OUT OF MEMORY，这还只是在batch_size设置为1的情况下。显卡状态如下：
![image](https://user-images.githubusercontent.com/43789460/190844362-73f7fb40-eea2-478c-9a3d-3302734b3a5b.png)

报错信息如下：
![image](https://user-images.githubusercontent.com/43789460/190844402-3d00269d-52ec-4d96-b39e-27cf41ad91af.png)

我的微调参数如下：
--pretrained_model_path IDEA-CCNL/Erlangshen-MegatronBert-1.3B 
--output_save_path /root/nlp/Gavin/Gaivn/nlp/PLM/Fengshenbang-LM/modelevaluation/output/predict.json 
--model_type huggingface-megatron_bert 
--data_dir /root/nlp/Gavin/数据/t_news/ 
--train_data train.json 
--valid_data dev.json 
--test_data test1.0.json 
--train_batchsize 1 
--valid_batchsize 64
--max_length 128 
--texta_name sentence 
--label_name label 
--id_name id 
--learning_rate 0.00002 
--weight_decay 0.1 
--num_labels 15 
--monitor val_acc 
--save_top_k 3 
--mode max 
--every_n_train_steps 100 
--save_weights_only True 
--dirpath /root/nlp/Gavin/Gaivn/nlp/PLM/Fengshenbang-LM/checkpoints/modelevaluation/tnews/ 
--filename model-{epoch:02d}-{val_acc:.4f} 
--max_epochs 7 
--check_val_every_n_epoch 1 
--val_check_interval 100 
--default_root_dir ./log/  
--accelerator gpu
--strategy ddp
--devices [0,1,2,3]
--num_workers 0

**Comment by GGGGGGXY** [comment]
strategy需要开启deepspeed，  单纯ddp无法微调1.3B模型，并且好像没有开启fp16，显存占用会比较高

**Comment by Gavingx** [comment]
> deepspeed

好的，谢谢。请问这块有什么比较好的文档可以参考吗？针对显存分析这块的文档

**Comment by GGGGGGXY** [comment]
> > deepspeed
> 
> 好的，谢谢。请问这块有什么比较好的文档可以参考吗？针对显存分析这块的文档

显存分析可以看这个文档，https://deepspeed.readthedocs.io/en/latest/memory.html  里面是一些关于deepspeed zero对于显存的分析，以及可以看到一部分zero的优化思路和方案。

**Comment by GGGGGGXY** [comment]
docker已经给到，可以通过下面方式使用
sudo docker run --runtime=nvidia --rm -itd --ipc=host --name fengshen fengshenbang/pytorch:1.10-cuda11.1-cudann8-devel
sudo docker exec -it fengshen bash
